File: frontend/src/api/menu.js
import http from './http'

export const getMenus = (params) => http.get('/menus', { params })
export const getMenu = (id) => http.get(`/menus/${id}`)
export const createMenu = (data) => http.post('/menus', data)
export const updateMenu = (id, data) => http.put(`/menus/${id}`, data)
export const deleteMenu = (id) => http.delete(`/menus/${id}`)

File: backend/routes/menu.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func menuRoutes(rg *gin.RouterGroup) {
	rg.GET("/menus", handlers.GetMenus)
	rg.GET("/menus/:id", handlers.GetMenu)
	rg.POST("/menus", handlers.CreateMenu)
	rg.PUT("/menus/:id", handlers.UpdateMenu)
	rg.DELETE("/menus/:id", handlers.DeleteMenu)

	rg.GET("/employees/:id/menu-permissions", handlers.GetEmployeeMenuPermissions)
	rg.PUT("/employees/:id/menu-permissions", handlers.SetEmployeeMenuPermissions)


}
